src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/ObjectPools/Extensions.cs

```diff
@@ -152,7 +152,7 @@ public static void ClearAndFree<T>(this ObjectPool<HashSet<T>> pool, HashSet<T>
         var count = set.Count;
         set.Clear();
 
-        if (count > Threshold)
+        if (count > Threshold && pool.TrimOnFree)
```

Comment:
>if (count > Threshold && pool.TrimOnFree) [](http://example.com/codeflow?start=8&length=41)

Should the other ClearAndFree methods be doing this too?

Reply:
probably :)  i can try to address in followup pr.